### PR TITLE
change: Use deterministic default store location when calling `meltano config set` with `--store=auto`

### LINKED ISSUE #6172: Unexpected behaviour when handling default values with Environments
### What is the current *bug* behavior?

Meltano does not take Environment inheritance into account when setting a value to its default in an Environment.

### What is the expected *correct* behavior?

Currently, when setting a setting value to it's default, Meltano will `unset` all values ("to fall back to default") rather than adding a new entry with the default value to `meltano.yml`. This behaviour was correct pre-environments, but does not take into account inheritance now that environments exist. The behaviour I would expect is that a new setting value would be added to the specific Environment with the default value, rather than the current behaviour that `unsets` values elsewhere in the inheritance chain.

### Steps to reproduce

```
# add example tap
meltano add extractor tap-gitlab

# configure example setting with non-default value in base
meltano --no-environment config tap-gitlab set fetch_merge_request_commits true

# configure same setting with default value in an Environment
meltano environment=dev config tap-gitlab set fetch_merge_request_commits false
```

The above results in the removal of the `True` setting in base, rather than the addition of a `False` setting value in the environment `dev`.

### Relevant logs and/or screenshots

Screenshot 2022-06-13 at 15 02 48

Bug behaviour - removed `True` setting in plugin base configuration and no additional `False` setting in Environment `dev`.
Expected behaviour:

Screenshot 2022-06-13 at 15 03 54

`true` retained in base and `false` added to Environment `dev` (even though it happens to be the default value).

### Further regression test
_Ensure we automatically catch similar issues in the future_

- [ ] Write additional adequate test cases and submit test results
- [ ] Test results should be reviewed by a person from the team

/label ~bug
/label ~"To Do"

### PR DESCRIPTION
**Pull request by kgpayne**
Closes #6172
- #6172

### CODE REVIEW

**Comment by aaronsteers** [comment]
@kgpayne - Is this still needed? What are next steps? I see tests are failing and I'm not sure of the remaining action items since #6116 is merged now.

**Comment by kgpayne** [comment]
@aaronsteers this is still required as [per this discussion](https://github.com/meltano/meltano/issues/6172#issuecomment-1206342549)in the parent issue. It is ready for review as previously marked 👍

> @tayloramurphy @aaronsteers I have confirmed the issue is still present in v2.3.0.
> 
> TLDR; we still need to fix this. Easiest way is to remove the trap door (and accept that users will in some cases store a copy of the default value in their meltano.yml). Harder way is make the trap door more intelligent, only calling unset if it is safe to do so (i.e. there are no other non-default values set elsewhere in the resolution hierarchy or other Environments). Easy way is probably a 4, harder way is an 8 or 12 (I don't know that we have a precedent for checking values in all stores, including all Environments, yet).
> 
> Root cause; this is still happening because we have a [trap door](https://github.com/meltano/meltano/blob/1c4bec902f42f2c96957fdda1d2e1484cb5c065f/src/meltano/core/settings_store.py#L1287) in the set method on the AutoStoreManager which checks if the new value is equal to the default value, and if so calls unset to avoid storing a redundant copy of the default value in meltano.yml. This made sense pre-environments, but causes issues now in the case above where a non-default base configuration cannot be overridden by a default value in an Environment.

Tests failures are unrelated to changes; one failing on a Dependency Review (this PR doesn't add or remove any dependencies, and the check now passes after the most recent merge to master 🤷‍♂️) and the other on a 'large yaml performance' test that regularly fails for the Windows+SQLite permutation (usually fixed by re-running). Coverage is down by 0.03% as I have removed tests associated with removed functionality.

**Review by edgarrmondragon — APPROVED** [review]

**Comment by aaronsteers** [comment]
@kgpayne - 

> ...the other on a 'large yaml performance' test that regularly fails for the Windows+SQLite permutation (usually fixed by re-running). 

Okay, thanks for clarifying the context on that. Let's see what we can do to remove/ignore this test if needed - but that can be in a separate PR.

**Comment by edgarrmondragon** [comment]
> Okay, thanks for clarifying the context on that. Let's see what we can do to remove/ignore this test if needed - but that can be in a separate PR. Do we have an issue on this or could you log one?
> 
> * For future ref, failure log is: https://github.com/meltano/meltano/runs/8143953509?check_suite_focus=true

@aaronsteers fwiw I logged here a few days ago: https://github.com/meltano/meltano/issues/6683

File: src/meltano/cli/cli.py [hunk 1]
```diff
@@ -94,6 +94,7 @@ def cli(  # noqa: WPS231
         selected_environment = None
         is_default_environment = False
         if no_environment or (environment and environment.lower() == "null"):
+            project.deactivate_environment()
```
[the remark this answers is not in the document]
Comment (kgpayne) [reply src/meltano/cli/cli.py:97]: Good question 😅 If I remember correctly, this actually resolved an issue in our test suite whereby a Project instance that previously had an active environment but was then passed the `—no-environment` flag was not deactiving its environment 🤔 Will confirm if this is still necessary 👍

[the remark this answers is not in the document]
Comment (kgpayne) [reply src/meltano/cli/cli.py]: Looks like it was just a transient bug. Removed the explicit deactivate and all is still well with tests 👍
